src/saveload/marker.rs

```diff
@@ -0,0 +1,213 @@
+//! Provides `Marker` and `MarkerAllocator` traits
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::hash::Hash;
+
+use shred::Resource;
+use {Component, DenseVecStorage, Entities, Entity, Join, ReadStorage, WriteStorage};
+
+use serde::ser::Serialize;
+use serde::de::DeserializeOwned;
+
+
+/// This trait should be implemetened by component which is gonna be used as marker.
+/// This marker should be set to entity that should be serialized.
```

Comment:
This sentense is unclear to me. Does it mean that implementing the trait marks it to be serialized?

Reply:
Implementing the trait makes the component a valid marker. Marked entities get serialized.
Which marker to use and which components to serialize is for user to decide.